enterprise/server/remote_execution/copy_on_write/copy_on_write.go

```diff
@@ -783,11 +779,16 @@ type Mmap struct {
 
 	Offset  int64
 	dataDir string
+	dirty   bool // dirty suffix, only used to compute path
+
+	// initMu is used only to prevent concurrent attempts to mmap the data.
+	initMu sync.Mutex
```

Comment:
@maggie-lou to answer your question from the other PR about whether this extra lock is really needed -

the old code for `ReadAt` looked like this:

```
initMap() // calls Lock / defer Unlock internally
// <-- at this location, we're not holding the lock - Unmap might be called here.
RLock()
defer RUnlock()
access m.data // unsafe access, since we might've been unmapped
```

in order to fix this, we could do:

```
Lock()
defer Unlock()
initMap() // don't lock internally
access m.data
```

But this will probably hurt concurrency since we're getting an exclusive lock. To fix this, we can do RLock() / RUnlock() in ReadAt, but have `initMap` internally have its own mutex (initMu). I don't think this will result in us being sloppier around concurrency, since initMu should only ever be needed inside initMap().